include/E57SimpleData.h

```diff
@@ -744,9 +744,6 @@ namespace e57
    using Data3DPointsData_d [[deprecated( "Will be removed in 4.0. Use Data3DPointsDouble." )]] =
       Data3DPointsData_t<double>;
 
-   extern template struct Data3DPointsData_t<float>;
-   extern template struct Data3DPointsData_t<double>;
-
```

Comment:
This change removes the whole reason for having explicit template instantiation in the cpp file.

What error is the original code giving you? What compiler are you using?

According to everything I looked at w.r.t. Windows said that the way I did it is correct and the CI tests [run properly](https://github.com/asmaloney/libE57Format/actions/runs/5345882696/job/14471326460) with the shared lib. Maybe there's some other issue that's showing up as a problem with this declaration?

Reply:
Quick follow-up. I just tried it locally with this compiler:

```
C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe
```

The test executable builds and runs without warnings/errors when using E57Format as a shared lib. It uses both `Data3DPointsFloat` and `Data3DPointsDouble`.

Reply:
Ok, so my original problem is that `Data3DPointsData_t<double>` and `Data3DPointsData_t<false>` are hidden in Linux shared library. If I added `E57_DLL` at line `747` I got the following Windows build errors:
```
D:\a\libE57Format\libE57Format\include\E57SimpleData.h(747): error C2220: the following warning is treated as an error
D:\a\libE57Format\libE57Format\include\E57SimpleData.h(747): warning C4910: 'e57::Data3DPointsData_t<float>': '__declspec(dllexport)' and 'extern' are incompatible on an explicit instantiation
D:\a\libE57Format\libE57Format\include\E57SimpleData.h(748): warning C4910: 'e57::Data3DPointsData_t<double>': '__declspec(dllexport)' and 'extern' are incompatible on an explicit instantiation
```
which is about one can't combine `extern` and `__declspec(dllexport)` at one line. And so I suggested to use `E57_DLL` in the `cpp` file.

Reply:
I will verify that the symbol is hidden on Linux tomorrow.